### PR TITLE
[Android] Fix clear button not working on Entry 

### LINKED ISSUE #7963: ClearButtonVisibility="WhileEditing" does not work in Android
### Description

Android: when pressing the X in the entry field, nothing happens.
Windows: works OK.

### Steps to Reproduce

1. Run the attached sample.
2. 
[MauiAppTestClearButtonVisibility.zip](https://github.com/dotnet/maui/files/8885684/MauiAppTestClearButtonVisibility.zip)

### Version with bug

6.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Pixel 5 API 30 Android 11.0

### Did you find any workaround?

No

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix clear button not working on Entry. The FlowDirection used when checking if the tapped point is inside the Clear button Rect was not correct. This PR adds changes to correct the issue.

fix-7963
 
Can use the .NET MAUI Gallery and the Entry samples to verify the changes.

### Issues Fixed

Fixes #7963

### CODE REVIEW

**Review by rachelkang — CHANGES_REQUESTED** [review]
Hmm it still appears to not be working for me

I added the following to Sandbox and the clear button still appears to not be functional:

``` 
    <VerticalStackLayout>

            <Entry
                HorizontalOptions="Center"
                WidthRequest="300"
                ClearButtonVisibility="WhileEditing"
                BackgroundColor="LightGray"
                Text="ClearButtonVisibility" />
            
    </VerticalStackLayout>
```
Am I missing something?

**Comment by jsuarezruiz** [comment]
> Hmm it still appears to not be working for me
> 
> I added the following to Sandbox and the clear button still appears to not be functional:
> 
> ```
>     <VerticalStackLayout>
> 
>             <Entry
>                 HorizontalOptions="Center"
>                 WidthRequest="300"
>                 ClearButtonVisibility="WhileEditing"
>                 BackgroundColor="LightGray"
>                 Text="ClearButtonVisibility" />
>             
>     </VerticalStackLayout>
> ```
> 
> Am I missing something?

Could you try it again?

```
<VerticalStackLayout>
      <Entry
          HorizontalOptions="Center"
          WidthRequest="300"
          ClearButtonVisibility="WhileEditing"
          BackgroundColor="LightGray"
          Text="ClearButtonVisibility" />
      <Entry
          HorizontalOptions="Center"
          WidthRequest="300"
          FlowDirection="RightToLeft"
          ClearButtonVisibility="WhileEditing"
          BackgroundColor="LightGray"
          Text="ClearButtonVisibility" />
      <Entry
          HorizontalOptions="Center"
          ClearButtonVisibility="WhileEditing"
          BackgroundColor="LightGray"
          Text="ClearButtonVisibility" />
      <Entry
          HorizontalOptions="Center"
          ClearButtonVisibility="WhileEditing" 
          FlowDirection="RightToLeft"
          BackgroundColor="LightGray"
          Text="ClearButtonVisibility" />
  </VerticalStackLayout>
```

**Comment by xlawswants** [comment]
> 
you can set the Text for "" or empty value ,You will see the phenomenon of the bug, the delete button will not show

**Comment by hartez** [comment]
The button does not disappear when the text length goes to zero during editing, and the button does not appear if the text length was zero when the Entry gained focus.

**Comment by hartez** [comment]
Okay, fixed it so the clear button status updates when the text changes. The button now appears when you add text to a blank entry, and disappears when the entry no longer has text.

File: src/Core/src/Platform/Android/EditTextExtensions.cs [hunk 1]
```diff
@@ -372,7 +360,7 @@ internal static bool HandleClearButtonTouched(this EditText? platformView, FlowD
 			if (motionEvent.Action != MotionEventActions.Up)
 				return false;
 
-			var x = motionEvent.GetX();
+			var x = motionEvent.RawX;
```
Comment (rachelkang) [inline src/Core/src/Platform/Android/EditTextExtensions.cs:363]: @hartez what is RawX vs GetX?

**Review by rachelkang — APPROVED** [review]
LGTM! Working now, yay :)

**Pull request merged**